keps/sig-windows/2578-windows-conformance/README.md

```diff
@@ -0,0 +1,418 @@
+
+#### Windows Service Proxying
+
+- Ability to support IPv6 and IPv4 services for accessing internal Linux pods from Windows pods over ClusterIP Endpoints.
+- Ability to support IPv6 and IPv4 services for accessing internal Linux pods from Windows pods over NodePort Endpoints.
+- Ability to support IPv6 and IPv4 services for accessing internal Windows pods from Linux pods over ClusterIP Endpoints.
+- Ability to support IPv6 and IPv4 services for accessing internal Windows pods from Linux pods over NodePort Endpoints.
+
+### Implementation
+
+This section details the implementation of a "tool" which verifies these clusters must allow for the following, differentiated set of verifications, specified above.
+
+- core conformance
+- hostProcess
+- activeDirectory
+- networkPolicy
+- serviceProxy
+
+#### Option 1 - Test tags
+
+One way to implement such a tool is to, specifically, target all available "ginkgo" tests in Kubernetes core itself, and run these one at a time (or concurrently).  Not we *do not* assert that these tests *must* run in parallel in order to comprise a conformance clusters (although this is often done in CI, its generally not a requirement of Kubernetes Conformance for linux clusters, either, since cluster capacity is not related to the functionality of a cluster). As an example, one such implementation of such a Conformance tool might be:
+
+```
+--ginkgo.focus=\\[Conformance\\]|\\[NodeConformance\\]|\\[sig-windows\\]|\\[sig-apps\\].CronJob --ginkgo.skip=\\[LinuxOnly\\]|\\[k8s.io\\].Pods.*should.cap.back-off.at.MaxContainerBackOff.\\[Slow\\]\\[NodeConformance\\]|\\[k8s.io\\].Pods.*should.have.their.auto-restart.back-off.timer.reset.on.image.update.\\[Slow\\]\\[NodeConformance\\]"
+```
+
+Currently, the defacto standard for windows testing resides at https://github.com/kubernetes-sigs/windows-testing, and we suggest this tag to users, however it should be noted that this does *not* conform to the specification in this document:
+
+- it triggers some tests which uses non `pause` images for validating functionality.
+- it doesn't implement the pod churn or emptyDir sharing tests.
+- it doesn't implement the *reboot* tests which are defined in this specification.
+
+The ability to specify HostProcess, NetworkPolicy, and other tests in this document would follow a similar tagging strategy, and any such tool could just invoke the existing Kuberentes end-to-end test binary with appropriate `focus` and `skip` arguments.
+
+#### Option 2 - Sonobuoy implementation for convenience
+
+Given the complexity of the above tagging scheme, we propose a concrete example of a binary program for Windows verification.
+
+```
+sonobuoy run --mode=windows-conformance --windows.hostProcess --windows.activeDirectory ...
+```
+
+In the above invocation, we verify
+
+- core conformance
+- hostProcess
+- activeDirectory 
```

Comment:
We would need to make sure there was a way to provide AD credentials for any GSMA test

Reply:
We should look at creating automation that sets up AD and any other required domain services so that we can have reproducible builds/tests/conformance.

Reply:
we have scripts that automate this for e2e tests, they wrapped to work with aks-engine but could be extracted out to run on any platform: https://github.com/kubernetes-sigs/windows-testing/tree/master/extensions  

https://testgrid.k8s.io/sig-windows-master-release#aks-engine-windows-containerd-serial-slow-master&include-filter-by-regex=GMSA